pkg/server/api/entry/v1/service.go

```diff
@@ -73,73 +73,86 @@ func (s *Service) GetEntry(ctx context.Context, req *entry.GetEntryRequest) (*ty
+func (s *Service) createEntry(ctx context.Context, e *types.Entry, outputMask *types.EntryMask) *entry.BatchCreateEntryResponse_Result {
 	log := rpccontext.Logger(ctx)
 
-	entry, err := api.ProtoToRegistrationEntry(e)
+	cEntry, err := api.ProtoToRegistrationEntry(e)
 	if err != nil {
 		log.WithError(err).Error("Invalid request: failed to convert entry")
-		return nil, status.Errorf(codes.InvalidArgument, "failed to convert entry: %v", err)
+		return &entry.BatchCreateEntryResponse_Result{
+			Status: api.CreateStatus(codes.InvalidArgument, "failed to convert entry: %v", err),
+		}
 	}
 
-	log = log.WithField(telemetry.SPIFFEID, entry.SpiffeId)
+	log = log.WithField(telemetry.SPIFFEID, cEntry.SpiffeId)
 
 	// Validates that there is no similar entry
 	res, err := s.ds.ListRegistrationEntries(ctx, &datastore.ListRegistrationEntriesRequest{
 		BySpiffeId: &wrappers.StringValue{
-			Value: entry.SpiffeId,
+			Value: cEntry.SpiffeId,
 		},
 		ByParentId: &wrappers.StringValue{
-			Value: entry.ParentId,
+			Value: cEntry.ParentId,
 		},
 		BySelectors: &datastore.BySelectors{
 			Match:     datastore.BySelectors_MATCH_EXACT,
-			Selectors: entry.Selectors,
+			Selectors: cEntry.Selectors,
 		},
 	})
 	if err != nil {
 		log.WithError(err).Error("Failed to list entries")
-		return nil, status.Errorf(codes.Internal, "failed to list entries: %v", err)
+		return &entry.BatchCreateEntryResponse_Result{
+			Status: api.CreateStatus(codes.Internal, "failed to list entries: %v", err),
+		}
 	}
 	if len(res.Entries) != 0 {
-		return nil, status.Error(codes.AlreadyExists, "entry already exists")
+		return &entry.BatchCreateEntryResponse_Result{
+			Status: api.CreateStatus(codes.AlreadyExists, "entry already exists"),
+		}
 	}
 
 	// Create entry
 	resp, err := s.ds.CreateRegistrationEntry(ctx, &datastore.CreateRegistrationEntryRequest{
-		Entry: entry,
+		Entry: cEntry,
 	})
 	if err != nil {
 		log.WithError(err).Error("Failed to create entry")
-		return nil, status.Errorf(codes.Internal, "failed to create entry: %v", err)
+		return &entry.BatchCreateEntryResponse_Result{
+			Status: api.CreateStatus(codes.Internal, "failed to create entry: %v", err),
+		}
 	}
 
 	if resp.Entry == nil {
 		log.Error("No entry returned")
-		return nil, status.Error(codes.Internal, "no entry returned")
+		return &entry.BatchCreateEntryResponse_Result{
+			Status: api.CreateStatus(codes.Internal, "no entry returned"),
+		}
 	}
 
 	tEntry, err := api.RegistrationEntryToProto(resp.Entry)
 	if err != nil {
 		log.WithError(err).Error("Unable to convert registration entry")
-		return nil, status.Errorf(codes.Internal, "unable to convert registration entry: %v", err)
+		return &entry.BatchCreateEntryResponse_Result{
+			Status: api.CreateStatus(codes.Internal, "unable to convert registration entry: %v", err),
+		}
 	}
 
 	applyMask(tEntry, outputMask)
 
-	return tEntry, nil
+	return &entry.BatchCreateEntryResponse_Result{
+		Status: api.CreateStatus(codes.OK, "ok"),
```

Comment:
I wonder if this could be a helper in the `api` package:

```
func OK() *types.Status {
    return &types.Status{Code: codes.OK, Message: codes.OK.String()},
}
```